xinference/_compat.py

```diff
@@ -84,8 +86,10 @@ class CreateChatCompletionOpenAI(BaseModel):
     n: Optional[int]
     parallel_tool_calls: Optional[bool]
     presence_penalty: Optional[float]
-    # we do not support this
-    # response_format: ResponseFormat
+    # FIXME schema replica error in Pydantic
+    # source: ResponseFormatJSONSchema in ResponseFormat
+    # use alias
+    # _response_format: Optional[ResponseFormat] = Field(alias="response_format")
```

Comment:
Is there any solution to this?

Reply:
It appears to be the same issue as [Xinference Issue #2032](https://github.com/xorbitsai/inference/issues/2032), and I have not yet found a solution.

Reply:
While the error is `Field name "schema" shadows a BaseModel attribute; use a different field name with "alias='schema'".`
I found https://github.com/openai/openai-python/blob/main/src/openai/types/shared_params/response_format_json_schema.py#L11 maybe the source of conflict. Because another declaration of json_schema https://github.com/openai/openai-python/blob/main/src/openai/types/shared/response_format_json_schema.py#L27
use schema_ instead of schema.

I will try to fix this by define ResponseFormat myself.

Reply:
Any update?

Reply:
works fine now.

Reply:
> works fine now.

https://github.com/xorbitsai/inference/pull/2391/commits/9d13391e103091760014b7dbce4fcdc17ba5815e